crates/core/tedge/src/cli/connect/c8y_direct_connection.rs

```diff
@@ -121,11 +122,42 @@ fn read_pvt_key(
 ) -> Result<rustls_0_19::PrivateKey, ConnectError> {
     // Become BROKER_USER to read the private key
     let _user_guard = user_manager.become_user(tedge_users::BROKER_USER)?;
-    let f = File::open(key_file)?;
+    match parse_pvt_key(key_file.clone()) {
+        Ok(key) => Ok(key),
+        Err(ConnectError::UnsupportedPvtKeyFormat) => match parse_pvt_rsa_key(key_file) {
+            Ok(key) => Ok(key),
+            Err(e) => Err(e),
+        },
+        Err(e) => Err(e),
+    }
```

Comment:
This can be more idiomatic.

```suggestion
parse_pvt_key(key_file.clone()).or_else(|_| parse_pvt_rsa_key(key_file))
```

The suggested code is not exactly identical, but this is on purpose: I would make no difference between the `UnsupportedPvtKeyFormat` and `RumqttcPrivateKe` error cases.

Reply:
done